client/state/themes/utils.js

```diff
@@ -211,6 +217,25 @@ export function isThemeFromWpcom( themeId ) {
 }
 
 /**
+ * Determine whether wpcom themes should be removed from
+ * a queried list of themes. For jetpack sites with the
+ * required capabilities, we hide wpcom themes from the
+ * list of locally installed themes.
+ *
+ * @param {Object} state Global state tree
+ * @param {number} siteId The Site ID
+ * @returns {boolean} true if wpcom themes should be removed
+ */
+export function shouldFilterWpcomThemes( state, siteId ) {
```

Comment:
Well I think that this check will die soon, so I could even live with it in selector for the remaining time.